Docs/AdvancedSetup.md

```diff
@@ -0,0 +1,45 @@
+# Advanced Setup
+
+Looking for examples that push CreateAPI to it's limits? Well look no further.
+
+<!-- Be sure to update this TOC when adding new sections! -->
+- [Using a different API Client](#using-a-different-api-client)
+
+## Using a different API Client
+
+While using [Get](https://github.com/kean/get) is the easiest way to get started with CreateAPI, you might want to integrate CreateAPI with a different client instead.
+
+To do this, there are two important steps that need following:
+
+1. Exclude the `Get` import in the generated paths
+2. Write a matching `Request` type
+
+### Exclude the `Get` import
+
+Using the [`paths.imports`](./ConfigOptions.md#pathsimports) option, override the default value (`["Get"]`) to omit the import:
+
+**.create-api.yaml**
+```yaml
+paths:
+  # Ensure that Get is not imported in generated source files
+  imports: []
+```
+
+### Write a matching `Request` type
+
+The generated code needs to initialize a type called `Request` to define all of the path parameters and response type. When importing Get, `Request` was already available to the generated code but after removing the import, this will no longer be the case.
+
+You should either add a new `Request` type to the module of the generated code, or import one from a different module (be sure to update `paths.imports` if so). The type should have the same interface as [Get's `Request` type](https://github.com/kean/Get/blob/899db7397eacddad384fc252d79b804c0801072c/Sources/Get/Request.swift#L11-L118):
+
+```swift
+struct Request<Response> {
+    var method: String = "GET"
+    var url: String
+    var query: [(String, String?)]?
+    var body: Encodable?
+    var headers: [String: String]?
+    var id: String?
+}
+```
```

Comment:
This would only work when we tweak how the generated Paths use `Request` (see discussion in #42)

Reply:
I added `imports: ["Get"]` parameter in the initial version as a basic way to exclude `Get` dependency. But the goal was to eventually provide a higher-level API for that, but I never got to actually adding it. If you are using a custom client, the `Request` type should probably be auto-generated by CreateAPI. The user could than provide extensions to their networking code to handle `Request`.

Reply:
> I added `imports: ["Get"]` parameter in the initial version as a basic way to exclude `Get` dependency. But the goal was to eventually provide a higher-level API for that, but I never got to actually adding it. If you are using a custom client, the `Request` type should probably be auto-generated by CreateAPI. The user could than provide extensions to their networking code to handle `Request`.

Good to know your thinking. I only document this as it's how we're approaching it in our project but I do agree, the current approach isn't ideal. 

I imagine that a `isUsingGet` (or `isUsingCustomAPIClient`) flag similar to `isGeneratingResponseHeaders` and `isNaiveDateEnabled` might be a good idea in the short term. Including `Request` in the generation is also a good idea, I didn't think about that (also I didn't realise that the struct was so simple until I added it here 😄)


> I can see people not desiring a `Request` object at all, using their CreateAPI generation for only the objects and string paths. _Not that I would do it myself_ but this seems safer to provide so that we don't "force" a dev to have a `Request` shim at all. However, this would be a big change for more discussion.

Interesting.. Although how about all of the other information (`method`, `query`, `body` etc?) I think `Request` is pretty lightweight on its own but open to other options. One of the less obvious benefits of `Request` is that it uses generics to associate the appropriate response type and I think people find that valuable?

Reply:
I just thought of that use case so it might not even be valid or something completely out of scope. I really like kean's original idea of CreateAPI defining `Request` itself and then devs extend off of that.